src/Microsoft.AspNetCore.ResponseCaching/Internal/DistributedResponseCacheStore.cs

```diff
@@ -20,40 +24,46 @@ public DistributedResponseCacheStore(IDistributedCache cache)
             _cache = cache;
         }
 
-        public object Get(string key)
+        public async Task<object> GetAsync(string key)
         {
             try
             {
-                return CacheEntrySerializer.Deserialize(_cache.Get(key));
+                return CacheEntrySerializer.Deserialize(await _cache.GetAsync(key));
             }
             catch
             {
-                return null;
+                return NullCompletedTask;
             }
         }
 
-        public void Remove(string key)
+        public Task RemoveAsync(string key)
         {
             try
             {
-                _cache.Remove(key);
+                return _cache.RemoveAsync(key);
```

Comment:
![image](https://cloud.githubusercontent.com/assets/2030323/18652881/ad3bb43a-7e8a-11e6-9dcc-001e98b055e4.png)

Updated the measurements after the change. The processing time is now favouring KVP in all cases but the difference is so small, it's within margin of error.